src/Microsoft.AspNet.Routing/Template/TemplateRoute.cs

```diff
@@ -203,19 +203,27 @@ public virtual string GetVirtualPath(VirtualPathContext context)
             // Validate that the target can accept these values.
             var childContext = CreateChildVirtualPathContext(context, values.AcceptedValues);
 
-            var path = _target.GetVirtualPath(childContext);
-            if (path != null)
+            var pathData = _target.GetVirtualPath(childContext);
+            if (pathData != null)
             {
                 // If the target generates a value then that can short circuit.
-                return path;
+                return pathData;
             }
 
             // If we can produce a value go ahead and do it, the caller can check context.IsBound
             // to see if the values were validated.
-            path = _binder.BindValues(values.AcceptedValues);
+            pathData = new VirtualPathData(this, _binder.BindValues(values.AcceptedValues));
```

Comment:
I think we should set use 'this' object for the Router of VirtualPathData, because the _target was not the router that created the virtual path.